src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -166,102 +248,345 @@ def read_nwbfile(
+                backend = "hdf5"
+        except:
+            raise RuntimeError(f"File {file_path} not found or not readable.")
+    elif file_path.is_dir():
+        try:
+            import zarr
+
+            with zarr.open(file_path, "r") as f:
+                backend = "zarr"
+        except:
+            raise RuntimeError(f"File {file_path} not found or not readable.")
+    else:
+        raise RuntimeError(f"File {file_path} not found or not readable.")
+    return backend
+
+
+def find_neurodata_type_from_backend(group, path="", result=None, neurodata_type="ElectricalSeries", backend="hdf5"):
     """
-    A RecordingExtractor for NWB files. This uses the NWB API to extract the traces and
-    the metadata and is called by the NwbRecordingExtractor factory.
+    Recursively searches for groups with the specified neurodata_type hdf5 or zarr object,
+    and returns a list with their paths.
+    """
+    if backend == "hdf5":
+        import h5py
+
+        group_class = h5py.Group
+    else:
+        import zarr
+
+        group_class = zarr.Group
+
+    if result is None:
+        result = []
+
+    for neurodata_name, value in group.items():
+        # Check if it's a group and if it has the neurodata_type
+        if isinstance(value, group_class):
+            current_path = f"{path}/{neurodata_name}" if path else neurodata_name
+            if value.attrs.get("neurodata_type") == neurodata_type:
+                result.append(current_path)
+            find_neurodata_type_from_backend(
+                value, current_path, result, neurodata_type, backend
+            )  # Recursive call for sub-groups
+    return result
+
+
+def extract_time_info_pynwb(electrical_series, samples_for_rate_estimation, load_time_vector=False):
+    """
+    Extracts the sampling frequency and the time vector from an ElectricalSeries object.
+    """
+    sampling_frequency = None
+    if hasattr(electrical_series, "rate"):
+        sampling_frequency = electrical_series.rate
+
+    if hasattr(electrical_series, "starting_time"):
+        t_start = electrical_series.starting_time
+    else:
+        t_start = None
+
+    timestamps = None
+    if hasattr(electrical_series, "timestamps"):
+        if electrical_series.timestamps is not None:
+            timestamps = electrical_series.timestamps
+            t_start = electrical_series.timestamps[0]
+
+    # TimeSeries need to have either timestamps or rate
+    if sampling_frequency is None:
+        sampling_frequency = 1.0 / np.median(np.diff(timestamps[:samples_for_rate_estimation]))
+
+    if load_time_vector and timestamps is not None:
+        times_kwargs = dict(time_vector=electrical_series.timestamps)
+    else:
+        times_kwargs = dict(sampling_frequency=sampling_frequency, t_start=t_start)
+
+    return sampling_frequency, times_kwargs
+
+
+def retrieve_electrodes_indices_from_electrical_series(open_file, electrical_series, backend="hdf5"):
```

Comment:
This seems to be the only place where zarr and hdf5 backend really diffe, right? I would add backend as a suffix to the name and make it a private method as well.